docs/src/userguide/config/event_config.rst

```diff
@@ -464,6 +464,134 @@ Example configuration for `Google Pub/Sub`_:
+                        {
+                            "name": "file_size_bytes",
+                            "type": "INTEGER",
+                            "mode": "NULLABLE"
+                        }
+                    ]
+                }
+
+.. option:: job_config.events.outputs[].type
+
+    Value: ``bq``
+
+    | **Runner**: Dataflow, Direct
+    | *Required*
+
+
+.. option:: job_config.events.outputs[].project STR
+
+    The ID of the project containing this table or ``None`` if the table reference is specified entirely by the |table|_ argument.
+
+    | **Runner**: Dataflow, Direct
+    | *Optional*
+
+.. _bigquery_output_dataset_config:
+.. option:: job_config.events.outputs[].dataset STR
+
+    The ID of the dataset containing this table or ``None`` if the table reference is specified entirely by the |table|_ argument.
+
+    | **Runner**: Dataflow, Direct
+    | *Optional*
+
+
+.. _bigquery_output_table_config:
+.. option:: job_config.events.outputs[].table STR
+
+    The ID of the table.
+    The ID must contain only letters ``a-z``, ``A-Z``, numbers ``0-9``, or underscores ``_``.
+    If |dataset|_ argument is ``None`` then the ``table`` argument must contain the entire table reference specified as: ``'DATASET.TABLE'`` or ``'PROJECT:DATASET.TABLE'``.
+
+    | **Runner**: Dataflow, Direct
+    | *Required*
+
+.. option:: job_config.events.outputs[].schema DICT
+
+    The schema to be used if the BigQuery table to write has to be created.
+    This should be specified as the string representing a dictionary in the form of:
+
+    .. code-block::
+
+        {
+            "fields": [
+                {
+                    "name": "<field name>",  # required
+                    "type": "<TYPE>",  # required
+                    "mode": "<MODE>",  # required
+                    "description": "<DESCRIPTION>"  # optional
+                }, ...
+            ]
+        }
+
+    Refer to the `BigQuery schema documentation <https://cloud.google.com/bigquery/docs/schemas#creating_a_json_schema_file>`_ for more information on defining a schema.
+
+    | **Runner**: Dataflow, Direct
+    | *Optional*
+
+
+.. option:: job_config.events.outputs[].create_disposition STR
+
+    A string describing what happens if the table does not exist. Possible values are:
+
+    * ``"CREATE_IF_NEEDED"``: create if does not exist.
+    * ``"CREATE_NEVER"``: fail the write if does not exist.
+
+    Defaults to ``"CREATE_IF_NEEDED"``.
+
+    | **Runner**: Dataflow, Direct
+    | *Optional*
+
+.. option:: job_config.events.outputs[].write_disposition STR
+
+    A string describing what happens if the table has already some data. Possible values are:
+
+    * ``"WRITE_TRUNCATE"``: delete existing rows.
+    * ``"WRITE_APPEND"``: add to existing rows.
+    * ``"WRITE_EMPTY"``: fail the write if table not empty.
+
+    Defaults to ``"WRITE_APPEND"``.
+
+    .. caution::
+
+        For streaming pipelines, ``"WRITE_TRUNCATE"`` can not be used.
```

Comment:
Good question, I had just copied this from beam's [docstrings](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/gcp/bigquery.py) 😆. I can't actually find more info on _why_ (I do see a `ValueError` will be raised if it happens). From this description in the [Beam docs](https://beam.apache.org/documentation/io/built-in/google-bigquery/#write-disposition):

> `BigQueryDisposition.WRITE_TRUNCATE`: Specifies that the write operation should replace an existing table. Any existing rows in the destination table are removed, and the new rows are added to the table.

I would _suspect_ it's because what do you consider an "existing table" when the job is streaming. I guess I'll just point to these beam docs for more info and just say a `ValueError` is raised lol